python/ray/serve/tests/test_deploy_app.py

```diff
@@ -1311,5 +1311,45 @@ def test_get_app_handle(client: ServeControllerClient):
     assert ray.get(handle_2.predict.remote("ADD", 2)) == "5 pizzas please!"
 
 
+@pytest.mark.parametrize("heavyweight", [True, False])
+def test_deploy_lightweight_multiple_route_prefix(
+    client: ServeControllerClient, heavyweight: bool
+):
+    """If user deploys a config that sets route prefix for a non-ingress deployment,
+    the deploy should fail.
+    """
+
+    config = {
+        "applications": [
+            {
+                "name": "default",
+                "import_path": "ray.serve.tests.test_config_files.world.DagNode",
+            }
+        ]
+    }
+    client.deploy_apps(ServeDeploySchema(**config))
+
+    def check():
+        assert requests.post("http://localhost:8000/").text == "wonderful world"
+        return True
+
+    wait_for_condition(check)
+
+    # Add route prefix for non-ingress deployment
+    config["applications"][0]["deployments"] = [{"name": "f", "route_prefix": "/"}]
+    if heavyweight:
+        # Trigger re-build of the application
+        config["applications"][0]["runtime_env"] = {"env_vars": {"test": "3"}}
+    client.deploy_apps(ServeDeploySchema(**config))
+
+    def check_failed():
+        s = serve.status().applications["default"]
+        assert s.status == ApplicationStatus.DEPLOY_FAILED
+        assert "Found multiple route prefixes" in s.message
+        return True
+
+    wait_for_condition(check_failed)
```

Comment:
can you run the check like 10x more times after the `wait_for_condition` to make sure the status doesn't oscillate?